components/infra-proxy-service/server/orgs.go

```diff
@@ -3,45 +3,29 @@ package server
 import (
 	"context"
 
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
-
 	"github.com/chef/automate/api/external/common/query"
 	secrets "github.com/chef/automate/api/external/secrets"
 	"github.com/chef/automate/api/interservice/infra_proxy/request"
 	"github.com/chef/automate/api/interservice/infra_proxy/response"
 	"github.com/chef/automate/components/infra-proxy-service/service"
 	"github.com/chef/automate/components/infra-proxy-service/storage"
+	"github.com/chef/automate/components/infra-proxy-service/validation"
 )
 
 // CreateOrg creates a new org
 func (s *Server) CreateOrg(ctx context.Context, req *request.CreateOrg) (*response.CreateOrg, error) {
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	if req.Id == "" {
-		s.service.Logger.Debug("incomplete create org request: missing org ID")
-		return nil, status.Error(codes.InvalidArgument, "must supply org ID")
-	}
-
-	if req.Name == "" {
-		s.service.Logger.Debug("incomplete create org request: missing org name")
-		return nil, status.Error(codes.InvalidArgument, "must supply org name")
-	}
-
-	if req.AdminUser == "" {
-		s.service.Logger.Debug("incomplete create org request: missing org admin user")
-		return nil, status.Error(codes.InvalidArgument, "must supply org admin user")
-	}
-
-	if req.AdminKey == "" {
-		s.service.Logger.Debug("incomplete create org request: missing org admin key")
-		return nil, status.Error(codes.InvalidArgument, "must supply org admin key")
-	}
+	// Validate all request fields are required
+	err := validation.New(validation.Options{
+		Target:          "org",
+		Request:         *req,
+		RequiredDefault: true,
+	}).Validate()
```

Comment:
Looking over the validation library internals a bit, which I had not seen before. Looks like `RequiredDefault` says that every field of the `Request` must be non-empty, is that correct?
Then I am not clear on the second loop inside `Validate()` which seems to be doing the same check on `Rules` (though I do not see you using `Rules` anywhere in the codebase). 
Ideally would be good if you could add a couple comment lines inside validation.go::Validate that addresses my questions.

Reply:
Yes, this validation library added to provide the validation out of the box. If `RequiredDefault` is set the all request field must be non-empty.


The second loop is the use case if the user wants to validate only specific fields to non-empty or other validations(TODO).

Name would be non-empty
```

err = validation.New(validation.Options{
		Target:          "org",
		Request:         *req,
		Rules: validation.Rules{
			"Name": []string{"required"}, // ip_address, email or multiple rules we can be add here but is in future scope.
		},
	}).Validate()
```


Will created further PRs and provides the doc & a couple of optimization for its uses. Thanks!